fontforge/svg.c

```diff
@@ -1553,9 +1553,11 @@ static SplineSet *SVGParsePath(xmlChar *path) {
 		end = skipcomma(end);
 		axisrot = strtod(end,&end)*FF_PI/180;
 		end = skipcomma(end);
-		large_arc = strtol(end,&end,10);
+		large_arc = *end != '0';
+		end++;
```

Comment:
`skipcomma()` is

```
static char *skipcomma(char *pt) {
    while ( isspace(*pt))++pt;
    if ( *pt==',' ) ++pt;
return( pt );
```
And therefore it reads leading spaces up to a potential comma and then stops. `strtol()` [eats leading whitespace](https://man7.org/linux/man-pages/man3/strtol.3.html). So it seems like this change would break a path fragment like `A 3.517, 3.517, 0, -0, 0, 7` because `*end` would just be a space here.

Reply:
I see, so I need to also eat leading whitespace if any before doing `*end`, right?

Reply:
This is just by inspection on my part, but yes I think that's right.

Reply:
Are we sure that this wouldn't mishandle some other valid input that begins with 0? I can't readily find whether the specification allows leading zeros in quantities, for example.

Reply:
I believe SVG arcs always have the full argument set and given where it's placed this code can only be parsing the flags, which are supposed to always be zero or one.

Reply:
Okay.

Reply:
Right, this only applies to a very narrow case, not every leading 0. These flags are always there, and repeated leading zeroes will still work.